Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -588,123 +531,126 @@ function Set-FrontDoorRequiredTLSVersion
-            
-            if (-not [System.Convert]::ToBoolean($isMinTlsVersionSetOnCustomDomain))
+           
+            try
             {
-                try
-                {
-                    Write-Host "Setting minimum required TLS version on custom domain : [$DomainName]" -ForegroundColor $([Constants]::MessageType.Info)
-                    Write-Host $([Constants]::SingleDashLine)
+                Write-Host "Setting minimum required TLS version on custom domain : [$DomainName]" -ForegroundColor $([Constants]::MessageType.Info)
+                Write-Host $([Constants]::SingleDashLine)
                     
-                    $secrets =  Get-AzFrontDoorCdnSecret -ResourceGroupName $resourceGroupName -ProfileName $resourceName
-                    $secretObj = $null
-
-                    foreach($secret in $secrets)
+                $customDomain = Get-AzFrontDoorCdnCustomDomain -CustomDomainName $DomainName  -ProfileName $resourceName -ResourceGroupName $resourceGroupName
+                if($customDomain -ne $null)
+                {
+                    if($customDomain.DomainValidationState -eq "Approved" -and $customDomain.DeploymentStatus -eq "Succeeded")
                     {
-                        if($secret.Name.Substring(37).Equals($DomainName))
+                        $tlsSettings = $customDomain.TlsSetting
+                        $tlsSettings.MinimumTlsVersion = $requiredMinTLSVersion
+                        $updatedDomain = Update-AzFrontDoorCdnCustomDomain -CustomDomainName $DomainName  -ProfileName $resourceName -ResourceGroupName $resourceGroupName -TlsSetting $tlsSettings
+                        if($updatedDomain -ne $null -and $updatedDomain.TlsSetting.MinimumTlsVersion -eq $requiredMinTLSVersion)
                         {
-                            $secretObj = $secret
+                            $frontdoorEndpoint.MinimumTlsVersion = $requiredMinTLSVersion
+                            $frontDoorsDomainsRemediated += $frontdoorEndpoint
+                            Write-Host "Minimum required TLS version for Front Door CDN Profiles has been set successfully for " [$DomainName]  -ForegroundColor $([Constants]::MessageType.Update)
+                            Write-Host $([Constants]::SingleDashLine)
                         }
+                    }                
+                    else{
+                    $frontDoorsDomainsSkipped += $frontdoorEndpoint
+                    $logResource = @{}
+                    $logResource.Add("DomainName",($_.DomainName))
+                    $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                    $logResource.Add("ResourceName",($_.ProfileName))
+                    $logResource.Add("Reason", "Warning! Domain Deployment Status and Domain Validation State must be Succeeded and Approved, Skipping this Front Door domain.")
+                    $logSkippedResources += $logResource
```

Comment:
if this is a special category, should it be displayed differently that, these resources are skipped because of Domain Validation State reason, not any other. 

It is getting clubbed with other and losing what we set out to do in first place. Only logging won't help but it should also be present in remediation summary in my opinion.